File: Backend/src/main/java/JourneySync/demo/service/ParticipantService.java
package JourneySync.demo.service;

import JourneySync.demo.dto.ParticipantResponse;
import JourneySync.demo.dto.request.CreateParticipantRequest;
import JourneySync.demo.dto.request.UpdateParticipantRequest;
import jakarta.transaction.Transactional;

import java.util.List;
import java.util.UUID;

public interface ParticipantService {

    @Transactional
    ParticipantResponse addParticipant(UUID tripId, CreateParticipantRequest request);

    List<ParticipantResponse> getParticipantsByTripId(UUID tripId);

    ParticipantResponse getParticipantById(UUID tripId, UUID participantId);

    @Transactional
    ParticipantResponse updateParticipant(
            UUID tripId,
            UUID participantId,
            UpdateParticipantRequest request
    );

    @Transactional
    void deleteParticipant(UUID tripId, UUID participantId);
}

File: frontend/src/services/participant.service.ts
import api from "./api";
import {
    Participant,
    CreateParticipantRequest,
    UpdateParticipantRequest,
} from "@/types/participant";
export const participantService = {

    async getParticipants(tripId: string): Promise<Participant[]> {

        const response = await api.get(
            `/trips/${tripId}/participants`
        );

        return response.data;

    },

    async createParticipant(
        tripId: string,
        request: CreateParticipantRequest
    ): Promise<Participant> {

        const response = await api.post(
            `/trips/${tripId}/participants`,
            request
        );

        return response.data;

    },

    async updateParticipant(
        tripId: string,
        participantId: string,
        request: UpdateParticipantRequest
    ): Promise<Participant> {

        const response = await api.put(
            `/trips/${tripId}/participants/${participantId}`,
            request
        );

        return response.data;

    },

    async deleteParticipant(
        tripId: string,
        participantId: string
    ): Promise<void> {

        await api.delete(
            `/trips/${tripId}/participants/${participantId}`
        );

    }

};
